app/lib/efile/nj/nj1040_calculator.rb

```diff
@@ -271,6 +273,54 @@ def calculate_line_58
         (@direct_file_data.fed_eic * 0.4).round
       end
 
+      def get_personal_excess(ssn, excess_type, threshold)
+        persons_w2s = @intake.state_file_w2s.all&.select { |w2| w2.employee_ssn == ssn }
+        return 0 unless persons_w2s.count > 1
+        return 0 if persons_w2s.any? { |w2| w2[excess_type] && w2[excess_type] > threshold }
+
+        excess_contribution_count = 0
+        total_contribution = 0
+
+        persons_w2s.each do |w2|
+          contribution = w2[excess_type] || 0
+          total_contribution += contribution
+          excess_contribution_count += 1 if contribution.positive?
+        end
+
+        if excess_contribution_count > 1 && total_contribution > threshold
+          return total_contribution - threshold
+        end
+        0
+      end
+
+      def calculate_line_59
+        total_excess = 0
+        excess_threshold = 179.78
```

Comment:
[pebble] can we store this number (and the one below) as consts on lines 6/7 with the other ones